main.go

```diff
@@ -2095,6 +2108,60 @@ func stopForwarding(cl *forward.ConnListener) {
 	}
 }
 
+func startMultiForwarding(device ios.DeviceEntry, mappings []string) {
+	var listeners []*forward.ConnListener
+
+	closeAllListeners := func() {
+		for _, l := range listeners {
+			l.Close()
+		}
+	}
+
+	for _, mapping := range mappings {
+		parts := strings.Split(mapping, ":")
+		if len(parts) != 2 {
+			closeAllListeners()
+			exitIfError("invalid mapping format", fmt.Errorf("expected hostPort:targetPort, got %s", mapping))
+		}
+		hostPort, err := strconv.ParseUint(parts[0], 10, 16)
+		if err != nil {
+			closeAllListeners()
+			exitIfError("invalid host port", err)
+		}
+		if hostPort < 1 {
+			closeAllListeners()
+			exitIfError("invalid host port", fmt.Errorf("port %d must be at least 1", hostPort))
+		}
+		targetPort, err := strconv.ParseUint(parts[1], 10, 16)
+		if err != nil {
+			closeAllListeners()
+			exitIfError("invalid target port", err)
+		}
+		if targetPort < 1 {
```

Comment:
there are multiple checks for hostPort's and targetPort's ranges, and that could be done in forward.Forward and eliminated from startMultiForwarding and from main.go. wdyt?